mailpoet/lib/API/JSON/v1/SendingQueue.php

```diff
@@ -92,7 +92,7 @@ public function add($data = []) {
       ]);
     }
 
-    $validationError = $this->validateNewsletter($newsletterEntity);
+    $validationError = $this->newsletterValidator->validate($newsletterEntity);
```

Comment:
Checking the code again after your comments, it occurred to me that there might be some redundancy here that I missed before. I'm saying this just but reading the code without testing, so I could be wrong. But now that we are using validate(), maybe we don't need to check for the subscriber limit anymore inside add() as validate already checks it (https://github.com/mailpoet/mailpoet/blob/3930d93c5a4a3ec5b27cfcc8d8d8b5fdf7bcd8f4/mailpoet/lib/API/JSON/v1/SendingQueue.php#L70)?

Reply:
I thought I made a separate comment about this but I don't think I actually ended up going into the detail I meant to, or I'm just not finding it. 

In any case, I'm finding myself wanting to go in the opposite direction and remove the subscriber check from the validator entirely. It feels wrong to have a `validateSubscriberLimit` method in the validator because it doesn't check anything specific to the newsletter itself. It's also nice to have a separate check and potential early return immediately inside `add` so we don't have to retrieve a newsletter first just to check if the subscriber limit has been hit. And if we did end up using this validator for the frontend side of things via an Ajax request, you would never be able to get meaningful validation errors anytime you would try to edit a newsletter when you're over the limit.

It really seems like an improper mixing of concerns and I think I'll remove it from the validator, though I'm certainly open to other ideas!

Reply:
Sounds good to me. Fair point.